### PR TITLE
Search correct language for Docusaurus docsearch

### PR DESCRIPTION
**Pull request by JoelMarcey**
## Motivation

Only search the current version of the docs. Right now there are duplicates because `next` is also searched.

https://github.com/facebook/Docusaurus/blob/d28b864a59fabeea45add8c090a13de7d0530de5/lib/core/Site.js#L140 does the replacement

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

eyes

## Related PRs

N/A

### CODE REVIEW

**Comment by endiliey** [comment]
Actually we should probably do it for languages too.

But the current state is that
https://github.com/algolia/docsearch-configs/blob/master/configs/docusaurus.json

Only has 'en' for lang and 'next' for version
So no matter which language or version we always only search on that lang & version.

I am currently away so I won't be able to submit PR for it.

File: website/siteConfig.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ const siteConfig = {
   algolia: {
     apiKey: '3eb9507824b8be89e7a199ecaa1a9d2c',
     indexName: 'docusaurus',
+    facetFilters: [ "lang:LANGUAGE", "version:VERSION" ],
```
[the remark this answers is not in the document]
Comment (JoelMarcey) [reply website/siteConfig.js]: Thanks!

**Comment by yangshun** [comment]
Is this PR being blocked by algolia/docsearch-configs#446?

**Comment by endiliey** [comment]
@yangshun. Yes, tbh we need to wait ~24 hours after that PR got merged so algolia crawler had finished crawling & we can test this

**Comment by s-pace** [comment]
👋 

I am currently trying to tailor you the best configuration in order to have something simple and easy to understand.

We are facing two majors issue regarding how we scrap your website:

1. There is no redirection from `https://docusaurus.io/docs/<lang>/ `to a document page like `https://docusaurus.io/docs/<lang>/installation`.
Could it be possible to have such redirection straight away?

2. Sitemap introduces pages with URL ending by ".html" for the english pages while the same page without this trailing ".html" are still available. This introduces some duplicates since we crawl pages with and without the trailing ".html" in order to discover non-english pages. Could it be possible to avoid these pages with the trailing ".html" from the sitemap?

An easier solution would be to extract the context of the page (version and lang) from the DOM itself. Could it be possible to add meta tags?

**Comment by JoelMarcey** [comment]
@s-pace Hi 👋 -- sorry this is causing so much trouble. But I am hoping if we solve this correctly for Docusaurus, then sites that use Docusaurus can benefit from what we learned.

> An easier solution would be to extract the context of the page (version and lang) from the DOM itself. Could it be possible to add meta tags?

I think we can probably do this, maybe. Just any `meta` tag? e.g.

```
<meta docLanguage = ...>
<meta docVersion = ...>
```

If we use meta tags:

1. Can we avoid the redirection problems you raised?
2. Would the sitemap issue be fixed too?
3. Would we need to change anything on the Algolia config side to enable this?

Thanks!

**Comment by endiliey** [comment]
@s-pace

1. I see two solutions of this. Either
A. Redirect https://docusaurus.io/docs/<lang>/ to any first available docs
B. Change the docsearch config start_url from https://docusaurus.io/docs/<lang>/ to https://docusaurus.io/docs/<lang>/installation so that it wont be 404 & algolia can crawl the site

Which do you prefer ?

2. In new version of Docusaurus, we provide user two options
1. Pages only accessible with .html extension
2. Pages are accessible with and without .html extension

If my understanding is right the solution for this is to provide a complete sitemap that only provide pages with or without .html pages (choose only one)

So the fix is to generate
1. Sitemap with only .html extension for option 1
2. Sitemap without .html extension for option 2

At https://docusaurus.io/sitemap.xml

**Comment by JoelMarcey** [comment]
> There is no redirection from `https://docusaurus.io/docs/<lang>/` to a document page like `https://docusaurus.io/docs/<lang>/installation`. Could it be possible to have such redirection straight away?

I created https://github.com/algolia/docsearch-configs/pull/453 as a short-term fix to this.

**Comment by JoelMarcey** [comment]
@s-pace I have a PR up for Docusaurus that creates a sitemap depending on whether a config option is set to allow extension-less URLs. So the sitemap will have `*.html` extensions if that config option is not set. If it is set, then the sitemap will not have any `*.html` extensions.

**Comment by yangshun** [comment]
@JoelMarcey Is this good to merge?

**Comment by JoelMarcey** [comment]
I am not sure we need this now. Or at least the whole thing. Do you see duplicate search results  when you search now?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by JoelMarcey** [comment]
I look at it again over the weekend and close, update or merge as appropriate.

**Comment by endiliey** [comment]
Actually the current state is that whichever our language is at, we always search for english language docs.

And whichever version we are currently, we always search for 'latest version'

https://github.com/algolia/docsearch-configs/blob/master/configs/docusaurus.json

**Comment by JoelMarcey** [comment]
Right. I think `latest` is good. We just want the language support.

**Comment by endiliey** [comment]
@joelmarcey
I'll patch your PR, test & merge it later.

**Comment by s-pace** [comment]
DocSearch PR merged:
```html
<!-- at the end of the HEAD -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css" />

<!-- at the end of the BODY -->
<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
<script type="text/javascript"> docsearch({
  apiKey: '3eb9507824b8be89e7a199ecaa1a9d2c',
  indexName: 'docusaurus',
  inputSelector: '### REPLACE ME ####',
  algoliaOptions: { 'facetFilters': ["lang:$LANG", "tags:$TAGS"] },
  debug: false // Set debug to true if you want to inspect the dropdown
});
</script>
```
- Add a search input in your page if you don't have any yet. Then update the inputSelector value in JS snippet to a CSS selector that targets your search input field.

- Replace `$LANG` with the lang you want to search on.
   The list of possible lang is hardcoded in the config.
   So as of today you have: `en`, `es-ES`, `ro`, `tr`, `zh-CN`

- Replace $TAGS with the tags you want to search on.
   The list of possible tags is hardcoded in the config.
   So as of today you have: blog, docs

**Comment by endiliey** [comment]
Search for chinese when on chinese
chinese

Search for espanol when on espanol
espanol

**Pull request merged**